contrib/internal/validationtest/validation_test.go

```diff
@@ -0,0 +1,279 @@
+}
+
+// RoundTrip adds the DD Tracer configuration environment and test session token to the trace request headers
+func (rt *testAgentRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
+	sessionTokenEnv, ok := os.LookupEnv("CI_TEST_AGENT_SESSION_TOKEN")
+	if !ok {
+		sessionTokenEnv = "default"
+	}
+	req.Header.Add("X-Datadog-Trace-Env-Variables", tracerEnv())
+	req.Header.Add("X-Datadog-Test-Session-Token", sessionTokenEnv)
+	return rt.base.RoundTrip(req)
+}
+
+func testAgentDetails() string {
+	testAgentHost, exists := os.LookupEnv("DD_TEST_AGENT_HOST")
+	if !exists {
+		testAgentHost = "localhost"
+	}
+
+	testAgentPort, exists := os.LookupEnv("DD_TEST_AGENT_PORT")
+	if !exists {
+		testAgentPort = "9126"
+	}
+	return fmt.Sprintf("%s:%s", testAgentHost, testAgentPort)
+}
+
+var (
+	testAgentConnection = testAgentDetails()
+	sessionToken        = "default"
+)
+
+func TestIntegrations(t *testing.T) {
+	// if _, ok := os.LookupEnv("INTEGRATION"); !ok {
+	// 	t.Skip("to enable integration test, set the INTEGRATION environment variable")
+	// }
+	integrations := []Integration{
+		memcachetest.New(),
+		dnstest.New(),
+	}
+
+	testCases := []struct {
+		name                   string
+		env                    map[string]string
+		integrationServiceName string
+	}{
+		{
+			"GlobalServiceConfigured",
+			map[string]string{
+				"DD_SERVICE": "Datadog-Test-Agent-Trace-Checks",
+			},
+			"",
+		},
+		{
+			"SpanAttributeSchemaV0",
+			map[string]string{
+				"DD_TRACE_SPAN_ATTRIBUTE_SCHEMA": "v0",
+				"DD_SERVICE":                     "Datadog-Test-Agent-Trace-Checks",
+			},
+			"",
+		},
+		{
+			"SpanAttributeSchemaV1",
+			map[string]string{
+				"DD_TRACE_SPAN_ATTRIBUTE_SCHEMA": "v1",
+				"DD_SERVICE":                     "Datadog-Test-Agent-Trace-Checks",
+			},
+			"",
+		},
+		{
+			"SpanAttributeSchemaV1WithIntegrationServiceName",
+			map[string]string{
+				"DD_TRACE_SPAN_ATTRIBUTE_SCHEMA": "v1",
+				"DD_SERVICE":                     "Datadog-Test-Agent-Trace-Checks",
+			},
+			"Datadog-Test-Agent-Trace-Checks-Override",
+		},
+	}
+
+	for _, ig := range integrations {
+		for _, tc := range testCases {
+			testName := fmt.Sprintf("contrib/%s/%s", ig.Name(), tc.name)
+
+			t.Run(testName, func(t *testing.T) {
+				sessionToken = fmt.Sprintf("%s-%d", testName, time.Now().Unix())
+				t.Setenv("CI_TEST_AGENT_SESSION_TOKEN", sessionToken)
+				// t.Setenv("DD_SERVICE", "Datadog-Test-Agent-Trace-Checks")
+				// loop through all our environment for the testCase and set each variable
+				for k, v := range tc.env {
+					t.Setenv(k, v)
+				}
+
+				// also include the testCase start options within the tracer config
+				tracer.Start(
+					tracer.WithAgentAddr(testAgentConnection),
+					tracer.WithHTTPClient(tracerHTTPClient()),
+				)
+				defer tracer.Stop()
+
+				if tc.integrationServiceName != "" {
+					componentName := ig.Name()
+					if componentName == "jmoiron/sqlx" {
+						componentName = "database/sql"
+					}
```

Comment:
Mind explaining this special case? Why should the `jmoiron/sqlx` contrib return `database/sql` as name?

Reply:
Removing this, I honestly cant remember but we can cross that bridge once those integrations are added.

Reply:
Please remove it